config/src/main/java/io/quarkus/ts/configmap/api/server/SmallRyeResource.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.ts.configmap.api.server;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import io.smallrye.config.SmallRyeConfig;
+import io.smallrye.config.SmallRyeConfigBuilder;
+
+@Path("/smallrye")
+public class SmallRyeResource {
+
+    private final SmallRyeConfig config;
+
+    public SmallRyeResource() {
+        config = new SmallRyeConfigBuilder()
+                .addDefaultSources()
+                .addDefaultInterceptors()
+                .withSecretKeys("secret.password")
+                .build();
+    }
+
+    @Path("/secret")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response getSecret() {
+        Response.ResponseBuilder result;
+        try {
+            result = Response.ok(config.getRawValue("secret.password"));
+        } catch (SecurityException ex) {
+            result = Response.serverError().entity(ex.getMessage());
```

Comment:
You could return DTO that contains both ex. message and return of
```Java
String secretValue = SecretKeys.doUnlocked(() -> {
    config.getValue("secret", String.class);
});

```
so we know it actually works when config is build manually in constructor.

Reply:
Added a separate test for this case